semgrep/semgrep/ignores.py

```diff
@@ -81,9 +81,12 @@ def _survives(self, path: Path) -> bool:
         return True
 
     def filter_paths(self, paths: Iterable[Path]) -> FrozenSet[Path]:
-        abspaths = (p.absolute() for p in paths if p.exists())
         return frozenset(
-            p for p in abspaths if self._survives(p) or p.samefile(self.base_path)
+            p
+            for p in paths
+            if p.exists()
+            and self._survives(p.absolute())
```

Comment:
If I'm not mistaken this is parsed as `(A and B) or C`, but shouldn't it be `A and (B or C)` ?

Reply:
you're right, fixed.